test/Microsoft.Azure.WebJobs.Host.EndToEndTests/WebJobs.Host.EndToEndTests.csproj

```diff
@@ -36,6 +36,15 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <LangVersion>default</LangVersion>
   </PropertyGroup>
+  <PropertyGroup>
+    <SignAssembly>true</SignAssembly>
+  </PropertyGroup>
+  <PropertyGroup>
+    <AssemblyOriginatorKeyFile>..\..\src\Common\PublicKey.snk</AssemblyOriginatorKeyFile>
```

Comment:
I think making it settable is fine (I've already started) -- was just trying to justify why i was hiding it before :-) 

@fabiocav -- it looks like the requirements for a LeaseId are that it must be a GUID. Do you remember why you kept the `HostInstanceId` a string in the Script repo? It seems like the easiest thing to do from a type-checking standpoint is just make it a GUID rather than a string.